src/pages/createInitiative.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Formik, Form } from 'formik';
+import Submit from '../components/Submit';
+import DropDown from '../components/DropDown';
+import TextField from '../components/Formik/TextField/TextField';
+import { navigate } from 'gatsby'
+
+interface Initiative {
+  category: string;
+  description: string;
+}
+
+const CreateInitiativePage: React.FC = () => {
+  const { t } = useTranslation();
+
+  let initiative: Initiative = {
+    category: '',
+    description: '',
+  }
+
+  const onSubmit = (values: Initiative, actions: any) => {
+    console.log(values)
+    console.log(actions)
+    setTimeout(() => {
+      fetch(`http://localhost:3000/api/v0/initiatives`, { // TODO: extract host into an env var
+        method: 'POST',
+        mode: 'cors',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(values), 
+        redirect: 'follow', 
+        referrerPolicy: 'no-referrer'
+      }).then(response => response.json()).then(body => {
+        navigate("/wallet");
+      });
+      actions.setSubmitting(false);
+    }, 1000);
+  };
+
+  return (
+    <>
+      <h1 className="mb-8">{t('Create an initiative')}</h1>
+      <p className="mb-4">
+        {t(
+          'Choose a category and a description for your initiative.'
+        )}
+      </p>
+      <label className="text-small font-bold block mb-1 text-gray-600">{t('Initiatives')}</label>
+        <DropDown // Hard-coded for now
+          items={[
+            { name: t("Education") },
+            { name: t("Elderly care") },
+            { name: t("Covid") }
+          ]}
+          onSelect={name => { initiative.category = name; console.log(initiative) }} // TODO: How to properly assign to component state?
```

Comment:
Not sure how to properly set the component's shared state so the drop down component can set the initiative's category, and the text field sets the description.

Reply:
React `useState()` hook, if you only need to change the value of this current "component", although it does also work for parent components if you pass in the values as props too.

Then for more complex things there's `useContext()` which allows you to write the context in the upper-most parent component... and then you can read from the context at any sub-child component (e.g. grand-grand-child component) without having to pass it through props at each state.